akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntities.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.internal
+
+import akka.actor.ActorLogging
+import akka.actor.Props
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ClusterShardingSerializable
+import akka.cluster.sharding.ClusterShardingSettings
+import akka.cluster.sharding.ShardRegion
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.ShardRegion.ShardId
+import akka.persistence.DeleteMessagesFailure
+import akka.persistence.DeleteMessagesSuccess
+import akka.persistence.DeleteSnapshotsFailure
+import akka.persistence.DeleteSnapshotsSuccess
+import akka.persistence.PersistentActor
+import akka.persistence.RecoveryCompleted
+import akka.persistence.SaveSnapshotFailure
+import akka.persistence.SaveSnapshotSuccess
+import akka.persistence.SnapshotOffer
+import akka.persistence.SnapshotSelectionCriteria
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final class EventSourcedRememberEntitiesStoreProvider(typeName: String, settings: ClusterShardingSettings)
+    extends RememberEntitiesShardStoreProvider {
+
+  override def shardStoreProps(shardId: ShardId): Props =
+    EventSourcedRememberEntitiesStore.props(typeName, shardId, settings)
+
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] object EventSourcedRememberEntitiesStore {
+
+  /**
+   * A case class which represents a state change for the Shard
+   */
+  sealed trait StateChange extends ClusterShardingSerializable {
+    val entityId: EntityId
+  }
+
+  /**
+   * Persistent state of the Shard.
+   */
+  final case class State private[akka] (entities: Set[EntityId] = Set.empty) extends ClusterShardingSerializable
+
+  /**
+   * `State` change for starting an entity in this `Shard`
+   */
+  final case class EntityStarted(entityId: EntityId) extends StateChange
+
+  case object StartedAck
+
+  /**
+   * `State` change for an entity which has terminated.
+   */
+  final case class EntityStopped(entityId: EntityId) extends StateChange
+
+  def props(typeName: String, shardId: ShardRegion.ShardId, settings: ClusterShardingSettings): Props =
+    Props(new EventSourcedRememberEntitiesStore(typeName, shardId, settings))
+}
+
+/**
+ * INTERNAL API
+ *
+ * Persistent actor keeping the state for Akka Persistence backed remember entities (enabled through `state-store-mode=persistence`).
```

Comment:
Pity this can't be implemented in typed. 
I guess as this is replacing a deprecated feature we can't remove yet we can't ask users to add a dep to typed sharding to get this

Reply:
I feel the same :) 
And yes, I don't want to introduce the dependency (plus the cognitive load of having to keeping track of both classic and typed actor specifics in the same component)